429 throttle

cmcavoy opened the pull request:
base changes for the throttle messaging. needs front end tlc.

toolness committed:
Fixed #100, removing unneeded code.

toolness committed:
added app test to ensure publishing works.

toolness committed:
added window.location test to publishing test.

Pomax committed:
made social media links relative to whatever protocol is being used

Pomax committed:
hotfix for social media introducing unsecure elements in https setting

Pomax committed:
Merge pull request #105 from Pomax/webpagemaker-issue-419-hotfix

hotfix for social media introducing unsecure elements in https setting

toolness committed:
Added externalreporter conditional script include to test/index.html for automated CI.

cmcavoy committed:
adding basic message passing, but need to remove the final confirmation pop up

toolness committed:
Yet another attempt to properly set base href for test suite in IE9.

Pomax commented on js/fc/ui/publish.js:
```diff
@@ -18,7 +18,7 @@ define(function() {
             // TODO: Put nicer error here.
             alert('Sorry, an error occurred while trying to get ' +
-                  'the page. :(');
+                  'the page. ');
```
there appears to be a rogue space left at the end now

Pomax commented on js/fc/ui/publish.js:
```diff
@@ -31,7 +31,7 @@ define(function() {
           if (err) {
             // TODO: Put nicer error here.
-            alert("Sorry, an error occurred while trying to publish. :(");
+            alert("Sorry, an error occurred while trying to publish. " + err.responseText);
```
unless our response text is a self contained descriptor, it's probably better to use "[...] trying to publish: " + err.responseText

toolness committed:
Updated slowparse.

toolness committed:
Updated slowparse.

toolness committed:
Allow Typekit to be disabled from querystring for testing purposes.

Right now Typekit appears to take a really long time to fail if
it's loaded on either an unsupported browser (like Opera, apparently)
or from a domain that isn't authorized. While we should generally fix
this to be more responsive for users on unsupported browsers, for now
it's making some unit/integration tests take a *lot* longer to run
than they ought to, so I'm committing this as a stopgap measure.

This is also probably related to #103, "Load Typekit asynchronously".

toolness committed:
Allow Typekit to be disabled from querystring for testing purposes.

Right now Typekit appears to take a really long time to fail if
it's loaded on either an unsupported browser (like Opera, apparently)
or from a domain that isn't authorized. While we should generally fix
this to be more responsive for users on unsupported browsers, for now
it's making some unit/integration tests take a *lot* longer to run
than they ought to, so I'm committing this as a stopgap measure.

This is also probably related to #103, "Load Typekit asynchronously".

Pomax committed:
errors are now a modal dialog

Pomax committed:
errors are now a modal dialog

Pomax committed:
added showing the dialog...

Pomax committed:
added error message back in

Pomax committed:
added publish suppression on error in main.js

Pomax committed:
forgot .length...

cmcavoy commented on slowparse:
```diff
@@ -1 +1 @@
-Subproject commit a33cce14c378cd9f78db3f6a4f9555a72931bfe2
+Subproject commit 541d71ad6cfaceceb69b73b1cc473941eb5feded
```
note sure what this changes...any idea what the changes to slowparse were?

Pomax replied:
test case fix for IE9.

cmcavoy commented on test/test-app.js:
```diff
@@ -2,7 +2,7 @@ require([], function() {
 
   function appTest(name, cb) {
-    var iframe = $('<iframe src="index.html"></iframe>');
+    var iframe = $('<iframe src="index.html?notypekit=1"></iframe>');
```
why don't we like typekit?

Pomax replied:
because this is for the test suite, which doesn't need to contact typekit every time we run over 9000 tests =)

cmcavoy commented:
consider this thing high fived!
